src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java

```diff
@@ -135,8 +134,8 @@ public TrackingXmlParserStream parse(final XMLStreamReader reader) throws XMLStr
             } else {
                 throw new IllegalStateException("Unsupported schema node type " + parentNode.getClass() + ".");
             }
-
-            read(reader, nodeDataWithSchema, reader.getLocalName(), tree);
+            final SchemaInferenceStack schemaIS = SchemaInferenceStack.of(codecs.getEffectiveModelContext());
```

Comment:
We do not need a stack - this class always begins with schemaroot - you can change parentNode type in constructor to effectivemodelcontext.

Reply:
Well, here is load whole model from XML with recursive. So for tracking all parts and creating correct SchemaTree it is needed.

Reply:
OK